app/gui/src/dashboard/components/dashboard/column/PathColumn.tsx

```diff
@@ -0,0 +1,235 @@
+import FolderIcon from '#/assets/folder.svg'
+import FolderArrowIcon from '#/assets/folder_arrow.svg'
+import { Button, Popover, Text } from '#/components/AriaComponents'
+import SvgMask from '#/components/SvgMask'
+import { useEventCallback } from '#/hooks/eventCallbackHooks'
+import { useCategoriesAPI, useCloudCategoryList } from '#/layouts/Drive/Categories/categoriesHooks'
+import type { AnyCloudCategory } from '#/layouts/Drive/Categories/Category'
+import { useUser } from '#/providers/AuthProvider'
+import { useSetExpandedDirectoryIds, useSetSelectedKeys } from '#/providers/DriveProvider'
+import type { DirectoryId } from '#/services/Backend'
+import { isDirectoryId } from '#/services/Backend'
+import { Fragment, useTransition } from 'react'
+import invariant from 'tiny-invariant'
+import type { AssetColumnProps } from '../column'
+
+/** A column displaying the path of the asset. */
+export default function PathColumn(props: AssetColumnProps) {
+  const { item, state } = props
+
+  const { virtualParentsPath, parentsPath } = item
+
+  const { getAssetNodeById } = state
+
+  const { setCategory } = useCategoriesAPI()
+  const setSelectedKeys = useSetSelectedKeys()
+  const setExpandedDirectoryIds = useSetExpandedDirectoryIds()
+
+  // Path navigation exist only for cloud categories.
+  const { getCategoryByDirectoryId } = useCloudCategoryList()
+
+  // Parents path is a string of directory ids separated by slashes.
+  const splittedPath = parentsPath.split('/').filter(isDirectoryId)
+  const rootDirectoryInPath = splittedPath[0]
+
+  const splittedVirtualParentsPath = virtualParentsPath.split('/')
+  // Virtual parents path is a string of directory names separated by slashes.
+  // To match the ids with the names, we need to remove the first element of the splitted path.
+  // As the first element is the root directory, which is not a virtual parent.
+  const virtualParentsIds = splittedPath.slice(1)
+
+  const { rootDirectoryId } = useUser()
+
+  const navigateToDirectory = useEventCallback((targetDirectory: DirectoryId) => {
+    const targetDirectoryIndex = finalPath.findIndex(({ id }) => id === targetDirectory)
+
+    if (targetDirectoryIndex === -1) {
+      return
+    }
+
+    const pathToDirectory = finalPath
+      .slice(0, targetDirectoryIndex + 1)
+      .map(({ id, categoryId }) => ({ id, categoryId }))
+
+    const rootDirectoryInThePath = pathToDirectory.at(0)
+
+    // This should never happen, as we always have the root directory in the path.
+    // If it happens, it means you've skrewed up
+    invariant(rootDirectoryInThePath != null, 'Root directory id is null')
+
+    // If the target directory is null, we assume that this directory is outside of the current tree (in another category)
+    // Which is the default, because path displays in the recent and trash folders.
+    // But sometimes user might delete a directory with it's whole content, and in that case we'll find it in the tree
```

Comment:
nit: `because the path`, `sometimes the user`, `it's` -> `its`, `because parent` -> `because the parent`